jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostGeneralSuite.scala

```diff
@@ -350,12 +348,21 @@ class XGBoostGeneralSuite extends FunSuite with TmpFolderPerSuite with PerTest {
     val modelPath = getClass.getResource("/model/0.82/model").getPath
     val model = XGBoostClassificationModel.read.load(modelPath)
     val r = new Random(0)
-    val df = ss.createDataFrame(Seq.fill(100000)(1).map(i => (i, i))).
+    var df = ss.createDataFrame(Seq.fill(100000)(1).map(i => (i, i))).
       toDF("feature", "label").repartition(5)
+    // since 0.82/model has 251 features and xgboost has enabled column check since 1.0.0
```

Comment:
I may not remember it correctly, what does it mean  by `since 0.82/model has 251 features`?

Reply:
Sorry, it's a type error. @trivialfis has analyzed the model and found the 0.82/model was trained with 251 features.

Reply:
@CodingCat @wbo4958 I tried to debug the failure of loading this classifier model in 0.82 release branch and master branch from Python.  In master branch it's impossible to load it.  So I switched back to 0.82 and the loaded model parameter is incorrect:

```
num_class:251, num_feature:1056964608
```
I'm not sure how is this model generated at the first place.  I regenerated a batch of models with 0.82 using: https://github.com/dmlc/xgboost/blob/master/tests/python/generate_models.py and pass all compatibility tests in 1.0 release branch, except for `Scikit-Learn ltr`.  Those tests have assertions for correct number of features and number of classes.

Reply:
@trivialfis since  the first 4 bytes of the model is the num_class saved by scala. So when you load the mode in C/C++ or python, maybe you need to skip the first 4 bytes.

Reply:
That's terrible .  We should remove it.

Reply:
no! it's intentional! you should never expect using any xgboost native api (python/c++/scala/java/r) to read a model saved by XGBoost-Spark

if you want to use python/c++/scala/java/r to access model trained by xgb-spark

you should do something like modelFromSpark.booster ,

Reply:
Got it,  reading the document for Spark, to export Spark model for other bindings one first need to access `nativeBooster`.  Thanks for the explanation.